### DISCUSSION TITLE
Ingress metrics

### DISCUSSION THREAD

**Discussion by jesussancheztellomm** [strimzi/strimzi-kafka-operator#6432] [Q&A] [answered]
Hi!
I am using ingress-nginx to expose my kafka cluster to the world but unfortunately i can't get any ingress metric. (number of requests, success rate, etc)

I've been running some tests, and it looks like the ingress-nginx controller pods are only exposing metrics when the ingress receives http requests but not for tcp requests.

Do you have any tip for get any basic metrics for the ingress?

Thanks in advance,
Jesús

**Comment by scholzj** [comment, marked as answer]
When Strimzi uses Ingress, it is using TLS passthrough. That basically hides the connection in Ingress as one long HTTPS connection which is not decrypted, so Ingress does not know what is inside (and even if it would, it would not know how to decode the Kafka protocol). So I think the only metrics you can get from Nginx for this will be around the connection itself.

**Reply by jesussancheztellomm** [reply]
Great thank you!

**Reply by jesussancheztellomm** [reply]
Hi again,
Is there any way to get per listener metrics? 
Looks like Confluent is able to offer some listener related metrics like:
```
kafka.server:type=socket-server-metrics,listener={listener_name},networkProcessor={#},name=connection-creation-rate
```

Thanks in advance,
Jesús

**Reply by scholzj** [reply]
You can configure the metrics as you wish or use the JMX directly if you need to. It is described in the docs.
